akka-projection-jdbc/src/test/scala/akka/projection/jdbc/JdbcProjectionSpec.scala

```diff
@@ -274,16 +275,28 @@ class JdbcProjectionSpec
     }
   }
 
+  private def assertOffsetIs(
+      expected: Long)(implicit offsetStore: JdbcOffsetStore[PureJdbcSession], projectionId: ProjectionId) = {
+    val offset = offsetStore.readOffset[Long](projectionId).futureValue.value
+    offset shouldBe expected
+  }
+
+  private def assertOffsetIsEmpty(
+      implicit offsetStore: JdbcOffsetStore[PureJdbcSession],
+      projectionId: ProjectionId) = {
+    withClue("check - offset is empty") {
+      val offsetOpt = offsetStore.readOffset[Long](projectionId).futureValue
+      offsetOpt shouldBe empty
+    }
+  }
```

Comment:
Good to extract these things. We could also consider extracting common things for Jdbc, Cassandra, Slick into a common base test.

On the other hand, our top priority should be to ship 1.0 of Projections and we can do that without test refactoring, and then make the refactoring when other more important things have been completed.